src/CloudTest.jl

```diff
@@ -79,6 +79,39 @@ function findOpenPorts(f, n)
     end
 end
 
+function _connect_with_timeout(host, port::Integer, timeout::Number)
+    s = TCPSocket()
+    t = Timer(_ -> close(s), timeout)
```

Comment:
I think this Timer constructor would make the parent task sticky:
```julia
julia> t = Threads.@spawn begin Timer(identity, 1) end
Task (done) @0x0000000148f939e0

julia> t.sticky
true
```
Maybe it's better to wrap it in a `fetch(@spawn(...))::Timer`

Reply:
I had something in the back of my mind about this, thanks for that!

Reply:
Done